tests/npm-app/src/endpoints/jwt.ts

```diff
@@ -0,0 +1,88 @@
+import { KJUR, KEYUTIL, ArrayBuffertohex } from 'jsrsasign'
+import jwt_decode from 'jwt-decode'
+import { Base64 } from 'js-base64'
+
+import * as ccf from '../types/ccf'
+
+interface JwtResponse {
+    userId: string
+}
+
+interface ErrorResponse {
+    msg: string
+}
+
+interface HeaderClaims {
+    kid: string
+}
+
+interface BodyClaims {
+    sub: string
+}
+
+export function jwt(request: ccf.Request): ccf.Response<JwtResponse | ErrorResponse> {
+    const authHeader = request.headers['authorization']
+    if (!authHeader) {
+        return unauthorized('authorization header missing')
+    }
+    const parts = authHeader.split(' ', 2)
+        if (parts.length !== 2 || parts[0] !== 'Bearer') {
+            return unauthorized('unexpected authentication type')
+        }
+    const token = parts[1]
+
+    // Extract header claims to select the correct signing key.
+    // We use jwt_decode() instead of jsrsasign's parse() as the latter does unnecessary work.
+    let headerClaims: HeaderClaims
+    try {
+        headerClaims = jwt_decode(token, { header: true }) as HeaderClaims
+    } catch (e) {
+        return unauthorized(`malformed jwt: ${e.message}`)
+    }
+    const signingKeyId = headerClaims.kid
+    if (!signingKeyId) {
+        return unauthorized('kid missing in header claims')
+    }
+
+    // Get the stored signing key to validate the token.
+    const keysMap = new ccf.TypedKVMap(ccf.kv['public:ccf.gov.jwt_public_signing_keys'], ccf.string, ccf.typedArray(Uint8Array))
+    const publicKeyDer = keysMap.get(signingKeyId)
+    if (publicKeyDer === undefined) {
+        return unauthorized(`token signing key not found: ${signingKeyId}`)
+    }
+    // jsrsasign can only load X.509 certs from PEM strings
```

Comment:
Curious: Why not store the certificate directly in PEM then?

Reply:
The JS code will go away and the C++ code will load DER certs directly which has lower parse overhead.